### DISCUSSION TITLE
Is x32dbg Supposed to Automatically Download Symbols From the Microsoft Symbol Server?

### DISCUSSION THREAD

**Discussion by hankinsohl** [x64dbg/x64dbg#3435] [Q&A]
In x32dbg preferences I've set the following:
Symbol Store: https://msdl.microsoft.com/download/symbols
Symbol Path: srv*d:\development\symbols

With the above settings, no symbols are downloaded and even the local store d:\development\symbols isn't used. when I attach the debugger to a process.

If I make this change
Symbol Path: d:\development\symbols

symbols are loaded from the local store d:\development\symbols when I attach to a process but the remote microsoft symbol server isn't used.

If I use the symdownload command, symbols are downloaded from the remote store and cached in the local store.  These symbols are subsequently loaded when I attach to a process for which they are relevant.

Is this the expected behavior?  Or is x32dbg supposed to automatically download symbols from the Microsoft symbol server as needed when a debugging session begins?  If I recall correctly, I think that the windows debuggers - windbg, and others - automatically download symbols when a debugging session starts.

**Comment by mrexodia** [comment]
It is not supposed to download automatically, you can trigger a download in the call stack tab and in the symbols tab (either per-module or for all modules).

**Comment by hankinsohl** [comment]
Thanks - good to know.

**Comment by hankinsohl** [comment]
Oh, by the way... if you format the symbol path using the preferences tab of the options dialog to look like this:

srv*d:\development\symbols*https://msdl.microsoft.com/download/symbols

the debugger crashes when opened.  When investigating this issue I tried the above and had to fix it by changing the CachePath variable in the debugger's ini file.
